File: backend/src/main/java/com/example/demo/dto/StockResponse.java
package com.example.demo.dto;

/**
 * Clase StockResponse del sistema de inventario.
 */
public class StockResponse {

    private String productSku;
    private String productName;
    private String locationCode;
    private String locationName;
    private Integer quantity;

    public StockResponse(
            String productSku,
            String productName,
            String locationCode,
            String locationName,
            Integer quantity
    ) {
        this.productSku = productSku;
        this.productName = productName;
        this.locationCode = locationCode;
        this.locationName = locationName;
        this.quantity = quantity;
    }

    public String getProductSku() { return productSku; }
    public String getProductName() { return productName; }
    public String getLocationCode() { return locationCode; }
    public String getLocationName() { return locationName; }
    public Integer getQuantity() { return quantity; }
}

File: AppAndroid/TFGApp/app/src/main/java/com/example/tfgapp/data/model/StockResponse.kt
// Archivo comentado: app/src/main/java/com/example/tfgapp/data/model/StockResponse.kt
// Explica la responsabilidad de esta clase y los pasos principales del flujo.

package com.example.tfgapp.data.model

// Modelo de datos usado para enviar o recibir información de la API.
data class StockResponse(
        val productSku: String,
        val productName: String,
        val locationCode: String,
        val locationName: String,
        val quantity: Int
)
